proposals/lip-0040.md

```diff
@@ -67,20 +75,35 @@ We define the following constants:
 
 _Figure 1: The general structure of the state sparse Merkle tree for a Lisk blockchain using two application-specific modules. The state root is the Merkle root. Each module defines its own module store. The keys of the leaf nodes start with the module IDs, so that each module subtree is separated from the others. Not all modules are depicted in this example._
 
+
+### State Root
+
 The state root is the root of a sparse Merkle tree (specified in [LIP 0039][lip-0039]), the state tree. Each registered module of the chain defines the key-value entries of its own module store. Values in the module store can be arbitrary byte arrays, but they typically correspond to data structures serialized according to [LIP 0027](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md). The keys of the corresponding leaf nodes are given by the concatenation of the module ID, store prefix, and the (32 bytes long) SHA-256 hash of the store key. Module IDs are bytes values of length `MODULE_ID_LENGTH`. Store prefixes are bytes values of length `STORE_PREFIX_ID_LENGTH`. This implies that a leaf-node key of the state tree has a fixed length of `MODULE_ID_LENGTH + STORE_PREFIX_ID_LENGTH + 32` bytes.
 
 In summary, for an entry in the global state store identified by module ID `moduleID`, store prefix `storePrefix`, store key `storeKey`, and store value `storeValue`, the corresponding leaf node is added to the state tree using `SMT.update(moduleID || storePrefix || hash(storeKey), hash(storeValue))`, where `||` indicates the bytes-concatenation operation. This leaf node will have the following properties:
 
-```
-leaf.key = moduleID || storePrefix || hash(storeKey),
-leaf.value = hash(storeValue),
+```python
+leaf.key = moduleID + storePrefix + SHA256(storeKey),
```

Comment:
Is there a specific reason why the hash function name is upper case in contrast to other function names we use? In Python3 it is also just `sha256`.

Reply:
no idea of the reason, but it is the standard in several other LIPs (as far back as LIP 3)

Reply:
that said, I think it would be completely clear to use `sha256` in the function